### PR TITLE
Add control CZ gate: CCZ

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

Add the controlled CZ gate: CCZ.
  ```
  >>> ccz = qml.CCZ(wires=[0, 1, 2])
  >>> matrix = ccz.compute_matrix()
  [[ 1  0  0  0  0  0  0  0]
   [ 0  1  0  0  0  0  0  0]
   [ 0  0  1  0  0  0  0  0]
   [ 0  0  0  1  0  0  0  0]
   [ 0  0  0  0  1  0  0  0]
   [ 0  0  0  0  0  1  0  0]
   [ 0  0  0  0  0  0  1  0]
   [ 0  0  0  0  0  0  0 -1]]
  ```

### CODE REVIEW

**Review by timmysilv — APPROVED** [review]
if there something to be done for `_add_cz` let's do that, otherwise lgtm!

File: tests/gate_data.py [hunk 1]
```diff
@@ -72,6 +72,9 @@
 Toffoli = math.diag([1 for i in range(8)])
 Toffoli[6:8, 6:8] = math.array([[0, 1], [1, 0]])
 
+CCZ = math.diag([1 for i in range(8)])
+CCZ[7, 7] = -1
```
[the remark this answers is not in the document]
Comment (rmoyard) [reply tests/gate_data.py:76]: Thanks !

File: pennylane/drawer/tape_mpl.py [hunk 2]
```diff
@@ -87,6 +87,7 @@ def _add_wirecut(drawer, layer, mapped_wires, op):
     ops.Toffoli: _add_cx,
     ops.MultiControlledX: _add_multicontrolledx,
     ops.CZ: _add_cz,
+    ops.CCZ: _add_cz,
```
[the remark this answers is not in the document]
Comment (rmoyard) [reply pennylane/drawer/tape_mpl.py:90]: Let me try that

[the remark this answers is not in the document]
Comment (timmysilv) [reply pennylane/drawer/tape_mpl.py:90]: confirmed offline it's normal because CCZ is symmetric across wires 👍

**Pull request merged**